### PR TITLE
4583-digitLength-should-be-renamed

### LINKED ISSUE #4583: digitLength should be renamed
Guess, what is the result of `333 digitLength`? It is `2`, of course ;-) You are supposed to use decimalDigitLength
We should rename this method to something less confusing, e.g. bytesCount.
While it is so confusing, it would be better to remove it without deprecation but it is used quite a lot so we shall use auto-deprecation

### PR DESCRIPTION
**Pull request by pavel-krivanek**
fixes #4583
rename Integer messages:
digitLength -> bytesLength
digitAt: -> bytesAt:
digitAt:put: -> bytesAt:put:
digitCompare: -> bytesCompare:

### CODE REVIEW

File: src/Deprecated80/Integer.extension.st [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		deprecated: 'Please use #byteAt:put: instead'
+		transformWith:
+			'`@receiver digitAt: `@statements1 put: `@statements2'
+		   -> '`@receiver byteAt: `@statements1 put: `@statements2'.
+
+	^ self byteAt: index put: value 
+]
+
+{ #category : #'*Deprecated80' }
+Integer >> digitCompare: arg [
```
Comment (dionisiydk) [inline src/Deprecated80/Integer.extension.st:28]: I did not know about this method. It seems it is same as #threeWayCompareTo: which was introduced for sorting. So I think we just need to use #threeWayCompareTo and do not keep a new one. But please alter the implementation to use optimized version from #digitCompare:.

Comment (dionisiydk) [reply src/Deprecated80/Integer.extension.st:28]: And It will move threeWayCompare to kernel

Comment (pavel-krivanek) [reply src/Deprecated80/Integer.extension.st:28]: The main difference is that it calls a primitive by default ;-)

Comment (dionisiydk) [reply src/Deprecated80/Integer.extension.st:28]: Yes. I mean let's change threeWayCompare to use digitCompare code. It will work faster

**Comment by Ducasse** [comment]
Oops there is a conflict.

File: src/Deprecated80/Integer.extension.st (hunk 1 above)
Comment (pavel-krivanek) [reply src/Deprecated80/Integer.extension.st:28]: I opened an issue for that: https://github.com/pharo-project/pharo/issues/4605

**Comment by pavel-krivanek** [comment]
only usual osx-32 failure now

**Pull request merged**